Fix errors with log table

triple0t opened the pull request:
## Description

This PR fixes the issues with the log table by cleaning it. The Log table is not an important table, so it's fine to clear it out once in a while.

It also updates the logging level to 'errors' for users who set theirs to 'everything'

More information here p1758792009531759-slack-C0602LR6D55

## Code review notes

NOTE: This PR will clean your Log table. 
Please back up the table if you still need it.

## QA notes

We can skip QA.

* Pull the PR
* Deactivate the MailPoet plugin
* Re-activate the MailPoet plugin
* Confirm empty logs at `/wp-admin/admin.php?page=mailpoet-logs`

## Linked PRs

_N/A_

## Linked tickets

_N/A_

## After-merge notes

_N/A_

## Tasks

- [x] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:fix-email-listing-reading-from-log-table)

_The latest successful build from `fix-email-listing-reading-from-log-table` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * One-time cleanup of the MailPoet log table to reduce size and improve performance.
  * Automatically adjusts overly verbose logging configurations to “Errors”; non-verbose settings remain unchanged.
  * Runs once per installation (no action required).

* **Documentation**
  * Added a changelog entry describing the log table cleanup update.

* **Tests**
  * Added integration tests validating logging-level adjustment and one-time log pruning.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

triple0t committed:
Add migration to update logging level and prune log table

This commit introduces a new database migration that updates the logging level from 'everything' to 'errors' for users who had previously set it to 'everything'. Additionally, it prunes the log table if it has not been pruned before, ensuring that unnecessary log entries are removed. Corresponding integration tests have been added to verify the migration's functionality.

triple0t committed:
Add changelog file

NeosinneR reviewed (dismissed):
Looks good. Can confirm that logs are pruned and settings updated. 

My only nitpick is missing existence check for tables, but looking at other migrations, we use that inconsistently. I'll leave it up to you whether to add it or not. 

| Before | After | 
| ---- | ---- | 
| <img width="1136" height="1345" alt="Image" src="https://github.com/user-attachments/assets/8d3d529e-8004-4c43-a828-f9555635e603" /> | <img width="937" height="416" alt="Image" src="https://github.com/user-attachments/assets/a1b87efc-2741-4377-81b3-747e88aae849" /> |

triple0t committed:
Add checks for table existence in migration methods

This update adds checks to ensure that the settings and log tables exist before attempting to update the logging level and prune the log table. This prevents potential errors during migration if the tables are not present.

triple0t committed:
Add a safe fallback if TRUNCATE is not permitted to avoid activation failures.

Some managed hosts restrict TRUNCATE (requires DROP privilege). If this throws, the migration (and potentially plugin activation) can fail. Wrap TRUNCATE in try/catch and fall back to DELETE; still set the flag when successful.
